apps/src/code-studio/components/SoundListEntry.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import Radium from 'radium';
+import * as color from "../../util/color";
+
+/**
+ * Component for a single sound tile in the Sound Library.
+ * Used in App Lab and Game Lab
+ */
+var SoundListEntry = React.createClass({
+  propTypes: {
+    assetChosen: React.PropTypes.func.isRequired,
+    soundMetadata: React.PropTypes.object.isRequired,
+    isSelected: React.PropTypes.bool.isRequired,
+    soundsRegistry: React.PropTypes.object.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      isPlaying: false
+    };
+  },
+
+  componentWillReceiveProps: function (nextProps) {
+    if (!nextProps.isSelected) {
+      this.setState({isPlaying: false});
+    }
+  },
+
+  clickSoundControl: function () {
+    if (this.state.isPlaying) {
+      this.props.soundsRegistry.stopAllAudio();
+      this.setState({isPlaying: false});
+    } else {
+      this.setState({isPlaying: true});
+      this.props.soundsRegistry.playURL(
+        this.props.soundMetadata.sourceUrl,
+        { onEnded: function () {
```

Comment:
leading brace should be on previous line. Could also use an arrow function, which will mean you don't need to bind to this.